src/core/src/runtime/itensor.cpp

```diff
@@ -160,7 +160,19 @@ void ITensor::copy_to(const std::shared_ptr<ov::ITensor>& dst) const {
 
     bool finish = false;
     for (size_t dst_idx = 0, src_idx = 0; !finish;) {
-        memcpy(dst_data + dst_idx, src_data + src_idx, src_strides[src_strides.size() - 1]);
+        if (get_element_type() == element::string) {
+            // in case string tensors, it needs to copy of new values for std::string objects
+            // memcpy is not suitable
+            auto dst_string = reinterpret_cast<std::string*>(dst_data + dst_idx);
+            auto src_string = reinterpret_cast<const std::string*>(src_data + src_idx);
+            size_t num_elements_stride = src_strides[src_strides.size() - 1] / element::string.size();
+            for (size_t idx = 0; idx < num_elements_stride; ++idx) {
```

Comment:
It looks like std::copy could be used instead custom loop, can you check?

Reply:
duplicate comment of this one: https://github.com/openvinotoolkit/openvino/pull/21244#discussion_r1409026413